### ISSUE TITLE
ssh on Mac OSX prints "communication with agent failed"

### ISSUE THREAD

**Issue by kouk** [tiwe-de/ssh-agent-filter#4]
I've been trying to use ssh-agent-filter on Mac OSX. I compiled it using boost 1.59.0 and everything seems fine. It communicates with my `ssh-agent`, and finds the correct keys. The problem is with communicating with my ssh client. 
Output from `ssh -vvv`:

```
[...]
debug3: send_pubkey_test
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 60
debug1: Server accepts key: pkalg ssh-rsa blen 279
debug2: input_userauth_pk_ok: fp SHA256:/UelsGUukQAAJNWhtAwzQcdkycmWZIeosd5GUdIiWrU
debug3: sign_and_send_pubkey: RSA SHA256:/UelsGUukQAAJNWhtAwzQcdkycmWZIeosd5GUdIiWrU
sign_and_send_pubkey: signing failed: communication with agent failed
```

In the `catch` block of `handle_client` if I print`std::current_exception().what()` I get:
```
"failed reading: Resource temporarily unavailable: unspecified iostream_category error"
```

Which is being thrown by boost in [`file_descriptor_impl::read`](http://www.boost.org/doc/libs/1_59_0/libs/iostreams/src/file_descriptor.cpp). The value of `errno` is `EAGAIN`. With `lldb` I found that the read happens [here](https://github.com/tiwe-de/ssh-agent-filter/blob/master/rfc4251.H#L187).

The funny thing is that if I step through the execution with the debugger, the problem disappears. But I don't understand why this is happening. The socket is blocking, and I don't see you setting any timeouts.

Any debugging tips?

**Comment by tiwe-de** [comment]
Perhaps the client socket inherits the non-blockingness of the listening socket on OS X? Try removing the O_NONBLOCK from it or at least print out its state.

**Comment by tiwe-de** [comment]
fixed in commit dc582af

**Comment by kouk** [comment]
Thanks a lot! It works with your latest commit. Great tool too.
